_commands/basics/sudo.md

```diff
@@ -3,22 +3,85 @@
 
 sudo
 ---
-TODO: Add documentation for this command by submitting a pull request.
-<!-- one line explanation would go here -->
 
-<!-- minimal example -->
+`sudo` allows a user to execute a command as a different user (with the default being the root user). `sudo` is useful when commands need to be performed with elevated privileges, such as installing system updates.
+
 ~~~ bash
-*short* example of command usage and output
+$    sudo apt-get update
 ~~~
 
 <!--more-->
 
 ### Useful Options / Examples
 
-#### Example command
+#### `sudo -ll`
+
+This allows a user to find out all of the root commands that they are able to execute and all the users they are able to execute these commands as. `sudo -l` could be used to do this, but for long lists, `sudo -ll` produces a cleaner output.
+
+~~~ bash
+user@computer-VirtualBox:~/c4cs.github.io$ sudo -ll
+[sudo] password for user: 
+Matching Defaults entries for user on computer:
+    env_reset, mail_badpass,
+    secure_path=/usr/local/sbin\:/usr/local/bin\:/usr/sbin\:/usr/bin\:/sbin\:/bin\:/snap/bin
+User user may run the following commands on computer:
+Sudoers entry:
+    RunAsUsers: ALL
+    RunAsGroups: ALL
+    Commands:
+	ALL
+~~~
+
+#### `sudo -u`
+
+This allows a user to carry out a command as another user. It will execute the command as the specified target user.
+
+~~~bash
+$    sudo -u serverUser rails db:drop
+[sudo] password for currentUser:
```

Comment:
`currentUser` -> `serverUser`, I believe?

Reply:
Thanks! As per the comment, I think for usage of sudo you enter the current user's password, not the target user? https://www.computerhope.com/unix/sudo.htm (under the description heading).

Reply:
Ah, yes. Great! Will merge this and give credit.